### PR TITLE
feat: add a new command to filter the content of Profile and Pset based on multiple package.xml

### PR DESCRIPTION
**Pull request by scolladon**
## What does this pull request contains? Explain your changes.

---

- [X] Added for new features.
- [ ] Changed for changes in existing functionality.
- [ ] Deprecated for soon-to-be removed features.
- [ ] Removed for now removed features.
- [X] Fixed for any bug fixes.
- [ ] Security in case of vulnerabilities.

## Explain your changes

---

Add a new command to have the ability to filter the content of a Profile and a Permission Set based on the content of multiple package.xml
The command takes a list of path of package.xml
The command takes a list of path where profile and permissionset should be located (recursively) to apply the filtering into them.
The command takes a list of userPermissions to keep. The actual behaviour is to remove every user permission not present in this white list.

## Does this close any currently open issues?

---

No issue related

- [ ] Jest test to check the fix is applied are added.

## Any particular element to being able to test locally

---

To test it you have to call the command `sfdx sgd:source:ppset -h`

## Any other comments?

---

This is WIP, no test case written yet and no traction from the community for this use case.
The current state of the code written is not acceptable

## Where has this been tested?

---

**Operating System:** Darwin Kernel Version 18.7.0

**Yarn version:** 1.22.10

**Node version:** v15.6.0

**sgd version:** 4.1.0

**git version:** 2.30.0

### CODE REVIEW

File: src/utils/parsingConstants.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+'use strict'
+const XML_HEADER = '<?xml version="1.0" encoding="UTF-8"?>\n'
+const XML_PARSER_OPTION = {
+  ignoreAttributes: false,
+  ignoreNameSpace: false,
+  arrayMode: true,
```
Comment (scolladon) [inline src/utils/parsingConstants.js:6]: This can fix some potential issue in very edge case

File: .prettierignore [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package-lock.json
+yarn.lock
```
Comment (scolladon) [inline .prettierignore:1]: This is an artefact when swapping from npm to yarn

File: src/utils/findInDir.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+            findInDir(filePath, filter, fileList);
+        } else if (filter.test(filePath)) {
+            fileList.push(filePath);
+        }
+    });
+
+    return fileList;
+};
+
+export { findInDir };
```
Comment (alexmodis) [inline src/utils/findInDir.ts:21]: my useless coin : remove ';' for coherence

File: src/commands/sgd/source/ppset.ts [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+  protected static requiresProject = true
+
+  public async run(): Promise<AnyJson> {
+    const project = await SfdxProject.resolve()
+    const projectJson = await project.resolveProjectConfig()
+    const basePath = project.getPath()
+    const packageDirectories = projectJson['packageDirectories'] as JsonArray
+    const defaultDir = packageDirectories.reduce(
+      (a, v) => (v['default'] === true ? (a = v['path']) : (a = a)),
```
Comment (alexmodis) [inline src/commands/sgd/source/ppset.ts:94]: readability

File: src/utils/findInDir.ts (hunk 3 above)
Comment (scolladon) [reply src/utils/findInDir.ts:21]: We need to have consistency between eslint and tslint
I will migrate everything to the new version of eslint (covering tslint also)

**Pull request closed without merging**